### PR TITLE
refactor: Update example app

### PR DESCRIPTION
**Pull request by Mercy811**
- Update example app to use the new APIs. Layout is the same as previous. 
- Update Flutter min version from `2.12.0` to `2.15.0` in order to support [name property of Enum ](https://api.flutter.dev/flutter/dart-core/EnumName/name.html)
   - [Segment](https://github.com/segmentio/analytics_flutter/blob/b7123b5ac2545f97146641a534e0be29b0117393/packages/core/pubspec.yaml#L7) min version is 2.19.2. So I feel it's ok to higher it.
- Note app life cycle events are not tracked. Have a separate [ticket](https://amplitude.atlassian.net/browse/AMP-94014) for it. 
image

Click all button and can see requests being sent successfully.
image
<details>
  <summary>Click to see paylaods!</summary>

<details>
<summary> Payload 1</summary>
```
{
  "api_key": "5b9a9510e261f9ead90865bbc5a7ad1d",
  "client_upload_time": "2024-02-28T21:00:08.325Z",
  "events": [
    {
      "event_type": "Dart Click",
      "user_id": "Xinyi",
      "device_id": "Xinyi's pixel",
      "time": 1709153978106,
      "event_properties": {},
      "user_properties": {},
      "groups": {},
      "group_properties": {},
      "platform": "Android",
      "os_name": "android",
      "os_version": "12",
      "device_brand": "google",
      "device_manufacturer": "Google",
      "device_model": "sdk_gphone64_arm64",
      "carrier": "T-Mobile",
      "language": "en",
      "ip": "$remote",
      "version_name": "1.0",
      "event_id": 4,
      "session_id": 1709153875403,
      "insert_id": "d744e8fb-a73c-4581-85dd-a706b1f6c0f3",
      "library": "amplitude-analytics-android\/1.14.0"
    },
    {
      "event_type": "$identify",
      "user_id": "Xinyi",
      "device_id": "Xinyi's pixel",
      "time": 1709154000792,
      "event_properties": {},
      "user_properties": {
        "$set": {
          "identify_test": "identify sent at 1709154000791",
          "key": "value"
        },
        "$add": {
          "identify_count": 1
        }
      },
      "groups": {},
      "group_properties": {},
      "platform": "Android",
      "os_name": "android",
      "os_version": "12",
      "device_brand": "google",
      "device_manufacturer": "Google",
      "device_model": "sdk_gphone64_arm64",
      "carrier": "T-Mobile",
      "language": "en",
      "ip": "$remote",
      "version_name": "1.0",
      "event_id": 5,
      "session_id": 1709153875403,
      "insert_id": "af1cb155-453b-4e6b-8ee0-8665e76837ee",
      "library": "amplitude-analytics-android\/1.14.0"
    },
    {
      "event_type": "$identify",
      "user_id": "Xinyi",
      "device_id": "Xinyi's pixel",
      "time": 1709154004792,
      "event_properties": {},
      "user_properties": {
        "$set": {
          "account": "acme"
        }
      },
      "groups": {
        "account": "acme"
      },
      "group_properties": {},
      "platform": "Android",
      "os_name": "android",
      "os_version": "12",
      "device_brand": "google",
      "device_manufacturer": "Google",
      "device_model": "sdk_gphone64_arm64",
      "carrier": "T-Mobile",
      "language": "en",
      "ip": "$remote",
      "version_name": "1.0",
      "event_id": 6,
      "session_id": 1709153875403,
      "insert_id": "cf8e4d2e-efde-477a-8f9d-597b7a555ef6",
      "library": "amplitude-analytics-android\/1.14.0"
    }
  ]
}
```
</details>
<details>
<summary>Payload 2</summary>
```
{
  "api_key": "5b9a9510e261f9ead90865bbc5a7ad1d",
  "client_upload_time": "2024-02-28T21:00:19.784Z",
  "events": [
    {
      "event_type": "$groupidentify",
      "user_id": "Xinyi",
      "device_id": "Xinyi's pixel",
      "time": 1709154015994,
      "event_properties": {},
      "user_properties": {},
      "groups": {
        "account": "acme"
      },
      "group_properties": {
        "$set": {
          "group key": "group value"
        }
      },
      "platform": "Android",
      "os_name": "android",
      "os_version": "12",
      "device_brand": "google",
      "device_manufacturer": "Google",
      "device_model": "sdk_gphone64_arm64",
      "carrier": "T-Mobile",
      "language": "en",
      "ip": "$remote",
      "version_name": "1.0",
      "event_id": 7,
      "session_id": 1709153875403,
      "insert_id": "bb783b51-38ee-4332-8d76-0b2fe869d6af",
      "library": "amplitude-analytics-android\/1.14.0"
    },
    {
      "event_type": "revenue_amount",
      "user_id": "Xinyi",
      "device_id": "Xinyi's pixel",
      "time": 1709154017846,
      "event_properties": {
        "$productId": "specialProduct",
        "$quantity": 2,
        "$price": 41.23
      },
      "user_properties": {},
      "groups": {},
      "group_properties": {},
      "platform": "Android",
      "os_name": "android",
      "os_version": "12",
      "device_brand": "google",
      "device_manufacturer": "Google",
      "device_model": "sdk_gphone64_arm64",
      "carrier": "T-Mobile",
      "language": "en",
      "ip": "$remote",
      "version_name": "1.0",
      "event_id": 8,
      "session_id": 1709153875403,
      "insert_id": "acd17c54-f017-4eaa-8bd3-b8ef24bf11cd",
      "library": "amplitude-analytics-android\/1.14.0"
    }
  ]
}
```
</details>

</details>

### CODE REVIEW

File: example/android/app/build.gradle [hunk 1]
```diff
@@ -27,6 +27,7 @@ apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 android {
+    namespace 'com.example.amplitude_flutte_example'
```
Comment (izaaz) [inline example/android/app/build.gradle]: nit: flutte**r**

File: lib/amplitude.dart [hunk 2]
```diff
@@ -50,7 +40,7 @@ class Amplitude extends _Amplitude {
       event.mergeEventOptions(options);
     }
 
-    return await _channel.invokeMethod("track", jsonEncode(event.toMap()));
```
Comment (izaaz) [inline lib/amplitude.dart:53]: Why is jsonEncode not needed anymore?

**Review by izaaz — APPROVED** [review]

**Review by qingzhuozhen — APPROVED** [review]

**Comment by yuhao900914** [comment]
Other questions:
1. why the branch is v4.x?
2. we'd better show the min support version, like the flutter version in the dev doc.
3. do we still need to support flutter web?

File: example/lib/main.dart [hunk 3]
```diff
@@ -2,4 +2,4 @@ import 'package:flutter/material.dart';
 
 import 'my_app.dart';
 
-void main() => runApp(const MyApp('a79cefed0b7076cf3998ef7578a18bf0'));
+void main() => runApp(const MyApp('5b9a9510e261f9ead90865bbc5a7ad1d'));
```
Comment (yuhao900914) [inline example/lib/main.dart]: shouldn't include any api key

File: example/android/app/build.gradle (hunk 1 above)
Comment (Mercy811) [reply example/android/app/build.gradle]: Nice catch! Update in https://github.com/amplitude/Amplitude-Flutter/pull/172/commits/5c4ad5998af32fb46d826192f946b66e3bc03d82

File: lib/amplitude.dart (hunk 2 above)
Comment (Mercy811) [reply lib/amplitude.dart:53]: `jsonEncode()` is what our Flutter SDK used before. But one big problem with JSON string is that, in the Android plugin, `getString()` casts `null` to a string `"null"`. `invokeMethod()` accepts both String and Map data type. So I believe it's easier to get the value at client side and also to debug.

File: example/lib/main.dart (hunk 3 above)
Comment (Mercy811) [reply example/lib/main.dart]: Updated in https://github.com/amplitude/Amplitude-Flutter/pull/172/commits/5c4ad5998af32fb46d826192f946b66e3bc03d82

**Comment by Mercy811** [comment]
@yuhao900914 

> why the branch is v4.x?

In case we need to release fix version for current Flutter SDK while developing the new Flutter SDK. Will merge v4.x into main with a breaking change.

> we'd better show the min support version, like the flutter version in the dev doc.

Could you elaborate on it? How and where to show it? I didn't find anything related it min version in dev doc.

> do we still need to support flutter web?

The pre-release version will not support web.

**Comment by yuhao900914** [comment]
- https://github.com/amplitude/Amplitude-Flutter/tree/v3.x?tab=readme-ov-file#compatibility

**Comment by Mercy811** [comment]
Thanks @yuhao900914, I updated readme at https://github.com/amplitude/Amplitude-Flutter/pull/172/commits/24780f787bc532f94060d92d5798ffef29cc4c6e

**Review by yuhao900914 — APPROVED** [review]

**Pull request merged**